src/backend/distributed/metadata/dependency.c

```diff
@@ -1080,6 +1080,54 @@ IsTableOwnedByExtension(Oid relationId)
 }
 
 
+/*
+ * ObjectAddressDependsOnExtension returns whether or not the object depends
+ * on an extension. It is assumed that an object having a dependency of type
+ * DEPENDENCY_AUTO_EXTENSION to an extension depends on that extension.
+ *
+ * If extensionAddress is not set to a NULL pointer, the function will write
+ * the extension address into extensionAddress.
+ */
+bool
+ObjectAddressDependsOnExtension(const ObjectAddress *target,
```

Comment:
Maybe combine this with IsObjectAddressOwnedByExtension, they only differ by `if (pg_depend->deptype == FOOBAR)` checks

Reply:
not applicable anymore